project/CelebornBuild.scala

```diff
@@ -872,3 +882,82 @@ trait FlinkClientProjects {
       )
   }
 }
+
+////////////////////////////////////////////////////////
+//                   MR Client                        //
+////////////////////////////////////////////////////////
+object MRClientProjects {
+
+  def mrClient: Project = {
+    Project("celeborn-client-mr", file("client-mr/mr"))
+      .dependsOn(CelebornCommon.common % "test->test;compile->compile")
+      .dependsOn(CelebornClient.client % "test->test;compile->compile")
+      .settings(
+        commonSettings,
+        libraryDependencies ++= Seq(
+          Dependencies.hadoopClientApi,
+          Dependencies.hadoopClientRuntime,
+          Dependencies.hadoopMapreduceClientApp
+        ) ++ commonUnitTestDependencies
+      )
+  }
+
+  def mrClientShade: Project = {
+    Project("celeborn-client-mr-shaded", file("client-mr/mr-shaded"))
+      .dependsOn(mrClient)
+      .settings(
+        commonSettings,
+
+        // align final shaded jar name with maven.
+        (assembly / assemblyJarName) := {
+          val extension = artifact.value.extension
+          s"${moduleName.value}_${scalaBinaryVersion.value}-${version.value}.$extension"
+        },
+
+        (assembly / test) := {},
+
+        (assembly / logLevel) := Level.Info,
+
+        // Exclude `scala-library` from assembly.
+        (assembly / assemblyPackageScala / assembleArtifact) := false,
+
+        (assembly / assemblyExcludedJars) := {
+          val cp = (assembly / fullClasspath).value
+          cp filter { v =>
+            val name = v.data.getName
+            !(name.startsWith("celeborn-") ||
+              name.startsWith("protobuf-java-") ||
+              name.startsWith("guava-") ||
+              name.startsWith("netty-") ||
+              name.startsWith("commons-lang3-") ||
+              name.startsWith("RoaringBitmap-"))
```

Comment:
lz4?

https://github.com/apache/incubator-celeborn/blob/8eba1b470e9cd54cf0cdcefc381433ce94d881a0/client-mr/mr-shaded/pom.xml#L76-L85

Reply:
hmm... it seems we don't need the `lz4-java` dependency for MR shaded client? cc @FMX

Reply:
It may be because in the hadoop MR environment, there is no lz4-java.

https://issues.apache.org/jira/browse/HADOOP-17292

Reply:
shall we also include `zstd-jni`? cc @FMX

Reply:
I think Hadoop has `zstd-jni` already. I'll make a test to verify this later.

Reply:
Looks like you have already added `zstd-jni`, I think it's OK.

Reply:
Hadoop uses the native zstd and does not use `zstd-ini`.

See https://issues.apache.org/jira/browse/HADOOP-13578